File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.Arrays;

class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        Arrays.sort(citations); 
        
        int k = 0; 
        for (int i = 0; i < citations.length; i++) {

            int h = citations.length - i; 


            if (citations[i] >= h) {

                if (h > k) {
                    k = h;
                }

                break; 
            }
        }
        
        answer = k;
        return answer;
    }
}

File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    answer = 0
    
    h_cita = []
    m = max(citations)
    
    for i in range(m + 1):
        n = 0 
        for citation in citations:
            if i <= citation:
                n += 1
        h_cita.append(n)
    
    for idx, h_index in enumerate(h_cita):
        if idx <= h_index:
            answer = idx  
    
    return answer  